[Bug] customSockopt is ignored for inbound UDP sockets on Linux

## Integrity requirements

- [x] I have read all the comments in the issue template and ensured that this issue meets the requirements.
- [x] I understand the configuration fields used in the reproducer and did not add unrelated options.
- [x] I provided a complete, self-contained unit reproducer and its output. Client/server configs are not applicable because the failure is isolated to the Linux socket-option helper before any protocol starts.
- [x] I searched open and closed issues for `customSockopt`, `inbound UDP`, and `SO_RCVBUF` and did not find a duplicate.
- [x] The problem reproduces on the latest pre-release, v26.7.28.

## Description

On Linux, `applyInboundSocketOptions` applies `CustomSockopt` only inside the `isTCPSocket(network)` branch. Consequently, every custom option is skipped for inbound `udp4` and `udp6` sockets, even when the entry explicitly has `"network": "udp"`.

The outbound implementation places the same generic `CustomSockopt` block outside its TCP-only branch and therefore works for UDP. The comment in the inbound block also says that the network prefix can be `udp`, which makes the current placement appear accidental.

Current source:

https://github.com/XTLS/Xray-core/blob/v26.7.28/transport/internet/sockopt_linux.go#L113-L197

Expected behavior: an inbound custom socket option is applied when `custom.Network` matches `udp4` or `udp6`.

Actual behavior: the option is silently skipped because the generic block is nested under `isTCPSocket(network)`.

This affects any Linux inbound UDP listener that needs a custom option such as `SO_RCVBUF`. A transport may independently change its buffer later, but that does not make the configured generic sockopt effective.

## Reproduction method

Checkout v26.7.28, add the following test as `transport/internet/sockopt_inbound_udp_test.go`, and run:

```bash
go test ./transport/internet -run '^TestInboundUDPCustomSockoptReceiveBuffer$' -count=1 -v
```

<details><summary>Complete unit reproducer</summary><pre><code>//go:build linux

package internet

import (
	"testing"

	"golang.org/x/sys/unix"
)

func TestInboundUDPCustomSockoptReceiveBuffer(t *testing.T) {
	fd, err := unix.Socket(unix.AF_INET, unix.SOCK_DGRAM, unix.IPPROTO_UDP)
	if err != nil {
		t.Fatal(err)
	}
	defer unix.Close(fd)

	before, err := unix.GetsockoptInt(fd, unix.SOL_SOCKET, unix.SO_RCVBUF)
	if err != nil {
		t.Fatal(err)
	}

	config := &SocketConfig{CustomSockopt: []*CustomSockopt{{
		System:  "linux",
		Network: "udp",
		Level:   "1",
		Opt:     "8",
		Value:   "8388608",
		Type:    "int",
	}}}
	if err := applyInboundSocketOptions("udp4", uintptr(fd), config); err != nil {
		t.Fatal(err)
	}

	after, err := unix.GetsockoptInt(fd, unix.SOL_SOCKET, unix.SO_RCVBUF)
	if err != nil {
		t.Fatal(err)
	}
	if after <= before {
		t.Fatalf("inbound UDP SO_RCVBUF was not changed: before=%d after=%d", before, after)
	}
}
</code></pre></details>

## Client config

<details><pre><code>Not applicable. The failure occurs in applyInboundSocketOptions before a client or protocol interaction is involved.</code></pre></details>

## Server config

<details><pre><code>Not applicable. The complete unit reproducer constructs the UDP socket and SocketConfig directly.</code></pre></details>

## Client log

<details><pre><code>Not applicable.</code></pre></details>

## Server log / test output

On unmodified v26.7.28:

<details><pre><code>=== RUN   TestInboundUDPCustomSockoptReceiveBuffer
    sockopt_inbound_udp_test.go:40: inbound UDP SO_RCVBUF was not changed: before=212992 after=212992
--- FAIL: TestInboundUDPCustomSockoptReceiveBuffer (0.00s)
FAIL
FAIL github.com/xtls/xray-core/transport/internet 0.004s
</code></pre></details>

As a control, moving the generic `CustomSockopt` block immediately outside the `isTCPSocket(network)` branch makes the same test pass:

<details><pre><code>=== RUN   TestInboundUDPCustomSockoptReceiveBuffer
--- PASS: TestInboundUDPCustomSockoptReceiveBuffer (0.00s)
PASS
ok github.com/xtls/xray-core/transport/internet 0.005s
</code></pre></details>

The minimal fix appears to be moving the existing generic block outside the TCP-only branch, matching `applyOutboundSocketOptions`. A regression test for both `udp4` and `udp6` would prevent recurrence.

--- a/infra/conf/transport_sockopt.go
+++ b/infra/conf/transport_sockopt.go
@@ -10,7 +10,7 @@ import (
 )
 
 type CustomSockoptConfig struct {
-	Syetem  string `json:"system"`
+	System  string `json:"system"`
 	Network string `json:"network"`
 	Level   string `json:"level"`
 	Opt     string `json:"opt"`
@@ -124,7 +124,7 @@ func (c *SocketConfig) Build() (*internet.SocketConfig, error) {
 
 	for _, copt := range c.CustomSockopt {
 		customSockopt := &internet.CustomSockopt{
-			System:  copt.Syetem,
+			System:  copt.System,
 			Network: copt.Network,
 			Level:   copt.Level,
 			Opt:     copt.Opt,

--- a/transport/internet/sockopt_linux.go
+++ b/transport/internet/sockopt_linux.go
@@ -169,41 +169,42 @@ func applyInboundSocketOptions(network string, fd uintptr, config *SocketConfig)
 				return errors.New("failed to set TCP_MAXSEG", err)
 			}
 		}
-		if len(config.CustomSockopt) > 0 {
-			for _, custom := range config.CustomSockopt {
-				if custom.System != "" && custom.System != runtime.GOOS {
-					errors.LogDebug(context.Background(), "CustomSockopt system not match: ", "want ", custom.System, " got ", runtime.GOOS)
-					continue
+	}
+
+	if len(config.CustomSockopt) > 0 {
+		for _, custom := range config.CustomSockopt {
+			if custom.System != "" && custom.System != runtime.GOOS {
+				errors.LogDebug(context.Background(), "CustomSockopt system not match: ", "want ", custom.System, " got ", runtime.GOOS)
+				continue
+			}
+			// Skip unwanted network type
+			// network might be tcp4 or tcp6
+			// use HasPrefix so that "tcp" can match tcp4/6 with "tcp" if user want to control all tcp (udp is also the same)
+			// if it is empty, strings.HasPrefix will always return true to make it apply for all networks
+			if !strings.HasPrefix(network, custom.Network) {
+				continue
+			}
+			level := 0x6 // default TCP
+			var opt int
+			if len(custom.Opt) == 0 {
+				return errors.New("No opt!")
+			} else {
+				opt, _ = strconv.Atoi(custom.Opt)
+			}
+			if custom.Level != "" {
+				level, _ = strconv.Atoi(custom.Level)
+			}
+			if custom.Type == "int" {
+				value, _ := strconv.Atoi(custom.Value)
+				if err := syscall.SetsockoptInt(int(fd), level, opt, value); err != nil {
+					return errors.New("failed to set CustomSockoptInt", opt, value, err)
 				}
-				// Skip unwanted network type
-				// network might be tcp4 or tcp6
-				// use HasPrefix so that "tcp" can match tcp4/6 with "tcp" if user want to control all tcp (udp is also the same)
-				// if it is empty, strings.HasPrefix will always return true to make it apply for all networks
-				if !strings.HasPrefix(network, custom.Network) {
-					continue
+			} else if custom.Type == "str" {
+				if err := syscall.SetsockoptString(int(fd), level, opt, custom.Value); err != nil {
+					return errors.New("failed to set CustomSockoptString", opt, custom.Value, err)
 				}
-				level := 0x6 // default TCP
-				var opt int
-				if len(custom.Opt) == 0 {
-					return errors.New("No opt!")
-				} else {
-					opt, _ = strconv.Atoi(custom.Opt)
-				}
-				if custom.Level != "" {
-					level, _ = strconv.Atoi(custom.Level)
-				}
-				if custom.Type == "int" {
-					value, _ := strconv.Atoi(custom.Value)
-					if err := syscall.SetsockoptInt(int(fd), level, opt, value); err != nil {
-						return errors.New("failed to set CustomSockoptInt", opt, value, err)
-					}
-				} else if custom.Type == "str" {
-					if err := syscall.SetsockoptString(int(fd), level, opt, custom.Value); err != nil {
-						return errors.New("failed to set CustomSockoptString", opt, custom.Value, err)
-					}
-				} else {
-					return errors.New("unknown CustomSockopt type:", custom.Type)
-				}
+			} else {
+				return errors.New("unknown CustomSockopt type:", custom.Type)
 			}
 		}
 	}

